tests/BUILD

```diff
@@ -1,6 +1,7 @@
-python_tests()
-
-python_sources(
-    name="tests0",
-    dependencies=["notebooks/sample.ipynb"]
+python_tests(
+    dependencies=[
+        "notebooks/sample.ipynb",
+        "//:root#jupyterlab",
```

Comment:
So, neither your test code not testbook has a dependency on jupyterlab. Pants can only infer dependencies from imports. If there is no such import, you need to tell Pants about the un-inferrable dependency.